src/main/java/org/generationcp/middleware/service/impl/study/StudyGermplasmServiceImpl.java

```diff
@@ -101,6 +104,54 @@ public long countStudyGermplasmByEntryTypeIds(final int studyId, final List<Stri
 
 	}
 
+	@Override
+	public Optional<StudyGermplasmDto> getStudyGermplasm(final int studyId, final int entryId) {
+		final StockModel stock = this.daoFactory.getStockDao().getById(entryId);
+		if (stock != null) {
+			if (studyId != stock.getProject().getProjectId()) {
+				throw new MiddlewareRequestException("", "invalid.entryid.for.study");
+			}
+			return Optional.of(new StudyGermplasmDto(stock));
+		}
+		return Optional.empty();
+	}
+
+	@Override
+	public StudyGermplasmDto replaceStudyGermplasm(final int studyId, final int entryId, final int gid, final String crossExpansion) {
+		// Check first that new GID is valid
+		final Germplasm newGermplasm = this.daoFactory.getGermplasmDao().getById(gid);
+		if (newGermplasm == null) {
+			throw new MiddlewareException("Invalid GID: " + gid);
+		}
+		final StockDao stockDao = this.daoFactory.getStockDao();
+		// Copy from old entry: entry type, entry #, entry code
```

Comment:
Do we still need to copy the seed source from old entry?

Reply:
The germplasm to be used is not coming for a germplasm list so there is no seed source to copy from. I don't think it's correct to retain the old seed source value, if any, as it's for a different germplasm